homeassistant/helpers/template.py

```diff
@@ -1380,6 +1381,34 @@ def state_attr(hass: HomeAssistant, entity_id: str, name: str) -> Any:
     return None
 
 
+def is_nominal(hass: HomeAssistant, entity_id: str | State | Iterable | None) -> bool:
+    """Test if an entity has a nominal state."""
+    # Entity is a non-existing object
+    if entity_id is None:
+        return False
+
+    # If entity is a string or object then make it a list first.
+    if isinstance(entity_id, (str, State)):
+        entity_id = [entity_id]
+
+    # Loop through entity items provided.
+    for entity in entity_id:
+        if isinstance(entity, str):
+            # Set value to state if string is an entity_id, otherwise to string itself.
+            value = (
```

Comment:
I'm not necessarily a fan of how this is handled, but I assume it was done to shorten the code?

This just makes it more readable IMO
```
if state_obj := _get_state(hass, entity):
   value = state_obj.state
else:
   value = entity
```

Reply:
I first had it pretty similar to that, but what happens then is that if one uses selectattr what is passed is a string with the value of that attribute. And then it would fail as <string>.state does not exist. Hence I need to check if it is a string with a . in it to determine if what is passed is a string entity_id or a value.

Reply:
first had it pretty similar to that, but what happens then is that if one uses selectattr what is passed is a string with the value of that attribute. And then it would fail as .state does not exist. Hence I need to check if it is a string with a . in it to determine if what is passed is a string entity_id or a value.

Reply:
But _get_state will return a state object or none to the walrus operator.  .state will always exist in that context, try it out.

Reply:
@ehendrix23 are you familiar with the walrus operator?

_get_state(hass, entity) which is none will be put into state_obj, which will cause the if statement to fail.  ANd thus only hit the else.

making your string that isn't an entity_id fall into value.

>If I do not check for the ., then if it is provided an entity string that does not exist it would then put the name of the entity-id in value and pass it whereas it shouldn't.
That's not true, that's not how the walrus operator works.

Reply:
I know how it works. I just tested as you asked:
```
    for entity in entity_id:
        if isinstance(entity, str):
            if state_obj := _get_state(hass, entity):
                value = state_obj.state
            else:
                value = entity
            # Set value to state if string is an entity_id, otherwise to string itself.
            # value = (
            #     getattr(_get_state(hass, entity), "state", None)
            #     if "." in entity
            #     else entity
            # )

            if value is None or value in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
                return False
        elif entity is None or entity.state in [STATE_UNAVAILABLE, STATE_UNKNOWN]:
            return False
```
This then fails test:
```{{ is_nominal("test.noobject") }}```
as test.noobject does not exist and now it just compares it as a string value which is incorrect and I'm assuming that is the place you asked me to change.

Reply:
if test.noobject doesn't exist, it should fail that...  Anything that's not in the state machine should fail the test.  I guess I'm confused as to why you think that should pass?

Reply:
Ah, I see what you're saying.  It clicked.  I think we should be searching for sluggified entity_id's then, not just strings with periods in them.   I believe there is a hass call that checks entity_id's.

Reply:
It fails the test. It returns TRUE instead of FALSE with that coding whereas it should return FALSE.
_get_state returns NONE and thus we go to the ELSE part of the IF statement.
There we then assign entity to value, thus value becomes "test.noobject".
When we then check, value is not NONE and it is not unavailable or unknown, hence it does not return FALSE.

Reply:
> Ah, I see what you're saying. It clicked. I think we should be searching for sluggified entity_id's then, not just strings with periods in them. I believe there is a hass call that checks entity_id's.

More than happy to use a call. Personally did not like checking for the period since if they string value provided contains a period we would consider it as an entity-id string and then fail it which would be wrong.

I can try to find the HASS call but open to help on it. :-)

Reply:
homeassistant.core.valid_entity_id

Reply:
Thx. Updated accordingly.  Tests pass. Pushed commit. :-)